### ISSUE TITLE
Is there a plan for `vscode` sdk to support Dialog API?

### ISSUE THREAD

**Issue by yiliang114** [microsoft/vscode#171138]
![image](https://user-images.githubusercontent.com/11473889/211966803-754e9e6f-bdf8-4eec-9b74-85a1f9c4ca39.png)

Now in the vscode sdk, there are already two APIs about dialog, but unfortunately, it is combined with the file system.
1. `showOpenDialog`
2. `showSaveDialog`

We cannot open a Dialog for the time being and use custom content. eg:  When we want to delete a file, we want to appear a Dialog to let the user confirm whether to delete it. In this case, the reminder in the lower right corner (use showInformationMessage) may not be very noticeable.

Is it possible to expose a new api, which the possible name is `vscode.window.showDialog`,Maybe we can fully refer to the parameter of `dialogService.show` for `showDialog`. 

I tried to search the list of issue, but I didn't find anything. Don't you have such a need?

If there is no plan to support it, I would like to know how does the official team consider this issue(that there is no support customized dialog api for extension)?

**Comment by alefragnani** [comment]
Hi @yiliang114 ,

Are you aware of that `showXXXMessage` API supports [modal messages](https://code.visualstudio.com/updates/v1_58#_details-for-modal-messages)? 

Wouldn’t be enough for your scenario?

Hope this helps

**Comment by yiliang114** [comment]
> Hi @yiliang114 ,
> 
> Are you aware of that `showXXXMessage` API supports [modal messages](https://code.visualstudio.com/updates/v1_58#_details-for-modal-messages)?
> 
> Wouldn’t be enough for your scenario?
> 
> Hope this helps

Thanks @alefragnani , this is useful. I found that this `showInformationMessage` API does have the way to use modal bullet boxes, and I looked at the source code. This turned out to be a function supported a long time ago.

Perhaps the most commonly used function of `showInformationMessage` has caused me some misunderstanding, causing me to think that it can only appear in the lower right corner, because the Message in the lower right corner and the Modal pop box look like two completely different effects.

Let me take a closer look at the document. At present, under my circumstances, it is enough for the time being.
